Spring-Boot/Spring-Boot-Problem/src/test/java/tw/waterball/judgegirl/springboot/problem/controllers/ProblemControllerTest.java

```diff
@@ -428,5 +430,66 @@ private void saveProblems(int... problemIds) {
         });
     }
 
+    @Test
+    void GivenOneProblemSavedAndTheLanguageEnvsHas_c_WhenProblemLanguageEnvsReplace_JAVA_with_c_ShouldLanguageEnvReplaced_JAVA() throws Exception {
+        int problemId = 1;
+        saveProblemAndTheLanguageEnvHas_C(problemId);
+
+        LanguageEnv languageEnv = createLanguageEnv(JAVA);
+        replaceProblemLanguageEnv(problemId, Language.C, languageEnv);
+
+        Problem actualProblem = mongoTemplate.findById(problemId, Problem.class);
+        ShouldProblemHasLanguageEnvs(actualProblem, JAVA);
+
+    }
+
+    @Test
+    void GivenOneProblemSavedAndLangEnvHas_C_WhenProblemReplace_JAVA_with_JAVA_AndGetThatProblem_ShouldProblemLangEnvHas_C_And_JAVA() throws Exception {
+        int problemId = 1;
+        saveProblemAndTheLanguageEnvHas_C(problemId);
+
+        LanguageEnv languageEnv = createLanguageEnv(JAVA);
+        replaceProblemLanguageEnv(problemId, Language.JAVA, languageEnv);
+
+        Problem actualProblem = mongoTemplate.findById(problemId, Problem.class);
+        ShouldProblemHasLanguageEnvs(actualProblem, Language.C, JAVA);
+
+    }
+
+
+    private void ShouldProblemHasLanguageEnvs(Problem actualProblem, Language... languages) {
```

Comment:
1. rename `problemShouldHaveLanguageEnv(Problem actualProblem, LanguageEnv expectedLangEnv)`
and improve the method's content please.

```java
LanguageEnv actualLangEnv = actualProblem.getLanguageEnv(expectedLangEnv.getName());
// then compare the equality of the two envs
```